core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c

```diff
@@ -951,14 +957,105 @@ wasi_path_remove_directory(wasm_exec_env_t exec_env, wasi_fd_t fd,
     return wasmtime_ssp_path_remove_directory(curfds, fd, path, path_len);
 }
 
+#if WASM_ENABLE_THREAD_MGR != 0
+static __wasi_timestamp_t
+get_timeout_for_poll_oneoff(const wasi_subscription_t *in,
+                            uint32 nsubscriptions)
+{
+    __wasi_timestamp_t timeout = UINT64_MAX;
+    uint32 i = 0;
+
+    for (i = 0; i < nsubscriptions; ++i) {
+        const __wasi_subscription_t *s = &in[i];
+        if (s->u.type == __WASI_EVENTTYPE_CLOCK
+            && (s->u.u.clock.flags & __WASI_SUBSCRIPTION_CLOCK_ABSTIME) == 0) {
+            timeout = min(timeout, s->u.u.clock.timeout);
+        }
+    }
+    return timeout == UINT64_MAX ? (__wasi_timestamp_t)-1
+                                 : timeout; /* returns timeout in ns */
```

Comment:
__wasi_timestamp_t is defined as uint64_t, `(__wasi_timestamp_t)-1` is also UINT64_MAX, seems we can just `return timeout;`

Reply:
done